### ISSUE TITLE
QOL: add `onRemove` to the `create-cluster` component to delete the k3d cluster

### ISSUE THREAD

**Issue by zack-is-cool** [defenseunicorns/uds-k3d#40]
### Is your feature request related to a problem? Please describe.
It would be cool to have the k3d cluster be deleted when doing a `zarf package remove` on this package. This also ties into UDS bundles, doing a `uds remove`

I think you'd just add an `onRemove` action to [this component here](https://github.com/defenseunicorns/uds-k3d/blob/main/zarf.yaml#L36C1-L64C90) to do a `k3d cluster delete...` command

### Describe the solution you'd like

- **Given** when this bundle is deployed via zarf or uds bundle
- **When** a user either types zarf package remove or uds remove containing this package
- **Then** the k3d cluster is deleted and everything is cleaned up neatly

### Describe alternatives you've considered
Do it myself in a uds runner task

**Comment by mjnagel** [comment]
I'm not entirely opposed to this, but in general it should be unnecessary and actually feels like a worse pattern than just destroying directly. On a fresh deploy of this package/a bundle including this the previous cluster will be deleted so for local dev subsequent deploys it's handled for you generally. 

For cleanup locally it feels arguably longer to type out a full zarf package/bundle remove vs just `k3d cluster delete <name>`. In the bundle scenario (and maybe even with just the package) you'd be waiting for things to uninstall from the cluster before the eventual cluster destruction which is added wasted time.

Interested in any thoughts here, understand the desire to reduce friction but I think package/bundle removes feel like the wrong way to destroy and this would encourage that as a bad pattern.

**Comment by jeff-mccoy** [comment]
yeah we actually used to do the remove, but did have some jankiness with that when deleting yourself, but generally prefer to use `k3d cluster delete` as the zarf action and readme currently say.
